addons/google_drive/models/google_drive.py

```diff
@@ -0,0 +1,185 @@
+            else:
+                raise UserError(_("Google Drive is not yet configured. Please contact your administrator."))
+        google_drive_client_id = IrConfigParameter.get_param('google_drive_client_id')
+        google_drive_client_secret = IrConfigParameter.get_param('google_drive_client_secret')
+        #For Getting New Access Token With help of old Refresh Token
+
+        data = werkzeug.url_encode(dict(client_id=google_drive_client_id,
+                                        refresh_token=google_drive_refresh_token,
+                                        client_secret=google_drive_client_secret,
+                                        grant_type="refresh_token",
+                                        scope=scope or 'https://www.googleapis.com/auth/drive'))
+        headers = {"Content-type": "application/x-www-form-urlencoded"}
+        try:
+            req = urllib2.Request('https://accounts.google.com/o/oauth2/token', data, headers)
+            content = urllib2.urlopen(req, timeout=TIMEOUT).read()
+        except urllib2.HTTPError:
+            if self.env.user._is_admin():
+                action_id = self.env.ref('base_setup.action_general_configuration').id
+                msg = _("Something went wrong during the token generation. Please request again an authorization code .")
+                raise RedirectWarning(msg, action_id, _('Go to the configuration panel'))
+            else:
+                raise UserError(_("Google Drive is not yet configured. Please contact your administrator."))
+        return json.loads(content).get('access_token')
+
+    @api.model
+    def copy_doc(self, res_id, template_id, name_gdocs, res_model):
+        google_web_base_url = self.env['ir.config_parameter'].get_param('web.base.url')
+        access_token = self.get_access_token()
+        # Copy template in to drive with help of new access token
+        request_url = "https://www.googleapis.com/drive/v2/files/%s?fields=parents/id&access_token=%s" % (template_id, access_token)
+        headers = {"Content-type": "application/x-www-form-urlencoded"}
+        try:
+            req = urllib2.Request(request_url, None, headers)
+            parents = urllib2.urlopen(req, timeout=TIMEOUT).read()
+        except urllib2.HTTPError:
+            raise UserError(_("The Google Template cannot be found. Maybe it has been deleted."))
+        parents_dict = json.loads(parents)
+        record_url = "Click on link to open Record in Odoo\n %s/?db=%s#id=%s&model=%s" % (google_web_base_url, self.env.cr.dbname, res_id, res_model)
+        request_url = "https://www.googleapis.com/drive/v2/files/%s/copy?access_token=%s" % (template_id, access_token)
+        headers = {'Content-type': 'application/json', 'Accept': 'text/plain'}
+        data_json = json.dumps({"title": name_gdocs, "description": record_url, "parents": parents_dict['parents']})
+        req = urllib2.Request(request_url, data_json, headers)
```

Comment:
@msh-odoo, there is already an issue to handle this exception.
Issue : https://www.odoo.com/web#id=647222&view_type=form&model=project.issue&menu_id=4120&action=609